spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractEndpoint.java

```diff
@@ -201,6 +201,11 @@ public final void stop(Runnable callback) {
 		}
 	}
 
+	@Override
+	public String getComponentType() {
+		return "endpoint";
```

Comment:
I think this is not correct: every each `AbstractEndpoint` must provide their proper implementation.